### PR TITLE
Migrating from bower to yarn

### PR DESCRIPTION
**Pull request by RishabhJain2018**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -40,9 +40,10 @@ fabfile/
 package-lock.json
 .sass-cache/
 node_modules/
-bower_components/
+bower_components
 frontend/dist/
 frontend/index.html
+yarn.lock
```
Comment (deshraj) [inline .gitignore:46]: We want to check in the yarn.lock file I think.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "karma start karma.conf.js"
```
Comment (deshraj) [inline package.json:7]: Might want to revert this change since we want to run karma tests.

Comment (deshraj) [reply package.json:7]: Also, there are a lot of deprecation warnings in the build console. Lets bump up the packages and try to use the latest ones (if possible). Would give us some speed up in terms of loading for sure.

File: docker/dev/nodejs/Dockerfile [hunk 3]
```diff
@@ -4,13 +4,12 @@ WORKDIR /code
...
 # Add dependencies
 ADD ./package.json /code
-ADD ./bower.json /code
 ADD ./gulpfile.js /code
 ADD ./.eslintrc /code
 ADD ./karma.conf.js /code
 
 # Install Prerequisites
-RUN npm install -g bower gulp
+RUN npm install -g yarn gulp
```
Comment (deshraj) [inline docker/dev/nodejs/Dockerfile:12]: Lets remove -g since we dont need it. Also, it throws error later if you are building from scratch

**Comment by deshraj** [comment]
Generally seems like most of the stuff is working fine except this error:

```
nodejs_1  | Required path doesn't exist: /code/bower_components/angular-moment-picker/dist/angular-moment-picker.css moment-picker
nodejs_1  | [10:40:52] 'vendorcss' errored after 817 μs
nodejs_1  | [10:40:52] Error: Promise rejected without Error
nodejs_1  |     at Domain.onError (/code/node_modules/async-done/index.js:49:15)
nodejs_1  |     at Object.onceWrapper (events.js:315:30)
nodejs_1  |     at emitOne (events.js:116:13)
nodejs_1  |     at Domain.emit (events.js:211:7)
nodejs_1  |     at Domain._errorHandler (domain.js:134:21)
nodejs_1  |     at process._fatalException (bootstrap_node.js:375:33)
```

**Comment by RishabhJain2018** [comment]
Closing this one!
